### PR TITLE
Fpet 312-Allocated Judge screen changes

### PR DESCRIPTION
**Pull request by GayatriDeviCh**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/FPET-312

### Change description ###

https://tools.hmcts.net/jira/browse/FPET-312

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ X] No
```

### CODE REVIEW

File: Jenkinsfile_CNP [hunk 1]
```diff
@@ -43,7 +43,9 @@ def secrets = [
       secret("citizen-user", "IDAM_CITIZEN_USERNAME"),
       secret("citizen-password", "IDAM_CITIZEN_PASSWORD"),
       secret('prl-citizen-frontend-idam-client-secret', 'CITIZEN_IDAM_CLIENT_SECRET'),
-      secret('filtered-court-ids', 'COURT_IDS_TO_FILTER')
+      secret('filtered-court-ids', 'COURT_IDS_TO_FILTER'),
+      secret('prl-rd-api-system-username', 'PRL_IDAM_REFDATA_USERNAME'),
+      secret('prl-rd-api-system-user-password', 'PRL_IDAM_REFDATA_PASSWORD')
```
Comment (hmcts-kodatala) [inline Jenkinsfile_CNP:48]: do we need secret  here ?

File: Jenkinsfile_nightly [hunk 2]
```diff
@@ -31,7 +31,9 @@ def secrets = [
     secret("rpa-notify-from-email", "RPA_NOTIFY_FROM_EMAIL"),
     secret("citizen-user", "IDAM_CITIZEN_USERNAME"),
     secret("citizen-password", "IDAM_CITIZEN_PASSWORD"),
-    secret('prl-citizen-frontend-idam-client-secret', 'CITIZEN_IDAM_CLIENT_SECRET')
+    secret('prl-citizen-frontend-idam-client-secret', 'CITIZEN_IDAM_CLIENT_SECRET'),
+    secret('prl-rd-api-system-username', 'PRL_IDAM_REFDATA_USERNAME'),
+    secret('prl-rd-api-system-user-password', 'PRL_IDAM_REFDATA_PASSWORD')
```
Comment (hmcts-kodatala) [inline Jenkinsfile_nightly:36]: do we need secret here ?

File: src/main/java/uk/gov/hmcts/reform/prl/services/RefDataUserService.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+            return listOfStaffResponse.stream()
+                .filter(response -> response.getStaffProfile().getUserType().equalsIgnoreCase(LEGALOFFICE))
+                .map(this::getDisplayEntry).collect(Collectors.toList());
+        }
+        return List.of(DynamicListElement.builder().build());
+    }
+
+    private DynamicListElement getDisplayEntry(StaffResponse staffResponse) {
+        String value = concat(staffResponse.getStaffProfile().getLastName(),"(").concat(staffResponse.getStaffProfile().getEmailId()).concat(")");
+        return DynamicListElement.builder().code(value).label(value).build();
```
Comment (hmcts-kodatala) [inline src/main/java/uk/gov/hmcts/reform/prl/services/RefDataUserService.java:89]: move constants to constant files

File: src/main/java/uk/gov/hmcts/reform/prl/services/gatekeeping/AllocatedJudgeService.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+    private AllocatedJudge mapAllocatedJudge(Map<String, Object> caseDataUpdated, DynamicList legalAdviserList,
+                                             RefDataUserService refDataUserService) {
+        AllocatedJudge.AllocatedJudgeBuilder allocatedJudgeBuilder = AllocatedJudge.builder();
+        if (null != caseDataUpdated.get("tierOfJudiciary")) {
+            allocatedJudgeBuilder.isSpecificJudgeOrLegalAdviserNeeded(YesOrNo.No);
+            allocatedJudgeBuilder.tierOfJudiciary(getTierOfJudiciary(String.valueOf(caseDataUpdated.get("tierOfJudiciary"))));
+        } else {
+            if (null != caseDataUpdated.get("isJudgeOrLegalAdviser")) {
+                if (AllocatedJudgeTypeEnum.judge.getId().equalsIgnoreCase(String.valueOf(caseDataUpdated.get("isJudgeOrLegalAdviser")))
```
Comment (hmcts-kodatala) [inline src/main/java/uk/gov/hmcts/reform/prl/services/gatekeeping/AllocatedJudgeService.java:45]: move constants to constant files

**Review by hmcts-kodatala — COMMENTED** [review]
pls check review comments

File: Jenkinsfile_CNP (hunk 1 above)
Comment (GayatriDeviCh) [reply Jenkinsfile_CNP]: fixed

File: src/main/java/uk/gov/hmcts/reform/prl/services/RefDataUserService.java (hunk 3 above)
Comment (GayatriDeviCh) [reply src/main/java/uk/gov/hmcts/reform/prl/services/RefDataUserService.java:89]: fixed

File: src/main/java/uk/gov/hmcts/reform/prl/services/gatekeeping/AllocatedJudgeService.java (hunk 4 above)
Comment (GayatriDeviCh) [reply src/main/java/uk/gov/hmcts/reform/prl/services/gatekeeping/AllocatedJudgeService.java]: fixed

**Pull request merged**